numpy/f2py/__init__.py

```diff
@@ -48,25 +56,48 @@ def compile(source,
         .. versionadded:: 1.11.0
 
     """
-    from numpy.distutils.exec_command import exec_command
     import tempfile
+    import shlex
+
     if source_fn is None:
-        f = tempfile.NamedTemporaryFile(suffix=extension)
+        f, fname = tempfile.mkstemp(suffix=extension)
+        # f is a file descriptor so need to close it
+        # carefully -- not with .close() directly
+        os.close(f)
```

Comment:
I think you can use `f = tempfile.NamedTemporaryFile(suffix=extension, delete=False)`, `f.close()`

Reply:
> New in version 2.6: The delete parameter.

Ah, that's why we didn't use it before.

Reply:
Looks like everything is going to pass in any case, so probably not worth changing.